Kernel/Helpers/MessageSender.cs

```diff
@@ -7,7 +7,7 @@ namespace iCSharp.Kernel.Helpers
 
 	public class MessageSender : IMessageSender
     {
-        private const string Delimeter = "<IDS|MSG>";
+        private const string Delimiter = "<IDS|MSG>";
```

Comment:
Can you remove this from here and put this in Kernel/Constants.cs and use it here and also in Shell.cs.